### PR TITLE
SDI-477: ✨ Remove topic service and tidy

### PR DESCRIPTION
**Pull request by petergphillips**
SDI-477: ✨ Remove topic service and tidy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -404,7 +404,7 @@ Start localstack with command:
 
 `docker-compose -f docker-compose-test.yml up localstack`
 
-Then run the `test-app` in your IDE from main class `HmppsTemplateKotlin` using Spring profiles `localstack` and `stdout`.
+Then run the `test-app` in your IDE from main class `HmppsTemplateKotlin` using Spring profiles `localstack`.
```
Comment (petergphillips) [inline README.md:407]: `stdout` profile removed - it is now the default

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueService.kt [hunk 2]
```diff
@@ -23,7 +23,7 @@ class MissingTopicException(message: String) : RuntimeException(message)
 
 open class HmppsQueueService(
   private val telemetryClient: TelemetryClient?,
-  hmppsTopicService: HmppsTopicService,
+  hmppsTopicFactory: HmppsTopicFactory,
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueService.kt:26]: this is reverting back to how it was before the standard and async versions.  With the topic service I couldn't seem to get the queue factory wired up before the health indicator stuff.

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt [hunk 3]
```diff
@@ -27,28 +29,23 @@ class HmppsSqsConfiguration {
...
+    hmppsTopicFactory: HmppsTopicFactory,
     hmppsQueueFactory: HmppsQueueFactory,
     hmppsSqsProperties: HmppsSqsProperties,
-  ) = HmppsQueueService(telemetryClient, hmppsTopicService, hmppsQueueFactory, hmppsSqsProperties)
+  ) = HmppsQueueService(telemetryClient, hmppsTopicFactory, hmppsQueueFactory, hmppsSqsProperties)
 
   @Bean
   @ConditionalOnMissingBean
-  @ConditionalOnExpression("'\${hmpps.sqs.reactiveApi:false}'.equals('false')")
+  @ConditionalOnWebApplication(type = SERVLET)
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt:43]: this looks a nicer way of deciding which resource to use - will be decided by spring instead

File: test-app-reactive/build.gradle.kts [hunk 4]
```diff
@@ -29,8 +29,9 @@ dependencies {
   testImplementation("org.mockito.kotlin:mockito-kotlin:4.1.0")
   testImplementation("org.mockito:mockito-inline:4.9.0")
   testImplementation("org.awaitility:awaitility-kotlin:4.2.0")
-  testImplementation("io.jsonwebtoken:jjwt:0.9.1")
-  testImplementation("com.github.tomakehurst:wiremock-standalone:2.27.2")
+  testImplementation("io.jsonwebtoken:jjwt-impl:0.11.5")
```
Comment (petergphillips) [inline test-app-reactive/build.gradle.kts:32]: update dependencies to new locations

File: test-app-reactive/src/main/resources/logback-spring.xml [hunk 5]
```diff
@@ -6,24 +6,12 @@
...
-
-    <springProfile name="!(dev || stdout)">
-        <appender name="consoleAppender" class="ch.qos.logback.core.ConsoleAppender">
-            <layout class="ch.qos.logback.classic.PatternLayout">
-                <Pattern>${LOG_PATTERN}</Pattern>
-            </layout>
-        </appender>
-        <appender name="logAppender" class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender"/>
-    </springProfile>
+    <appender name="consoleAppender" class="ch.qos.logback.core.helpers.NOPAppender"/>
```
Comment (petergphillips) [inline test-app-reactive/src/main/resources/logback-spring.xml:9]: now just use the stdout version as app not intended to be deployed

File: test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/JwtAuthHelper.kt [hunk 6]
```diff
@@ -43,7 +44,7 @@ class JwtAuthHelper {
       .setSubject(subject)
       .addClaims(claims)
       .setExpiration(Date(System.currentTimeMillis() + expiryTime.toMillis()))
-      .signWith(RS256, keyPair.private)
+      .signWith(keyPair.private, SignatureAlgorithm.RS256)
```
Comment (petergphillips) [inline test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/JwtAuthHelper.kt:47]: new version had changed order of `signWith`

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**